tests/conftest.py

```diff
@@ -1100,3 +1106,81 @@ def ios_store(mocker: plugin.MockerFixture, db_engine_path: str) -> models.IosSt
         bundle_id="com.example.ios", application_name="Example iOS App"
     )
     return asset
+
+
+@pytest.fixture
+def nmap_agent_def() -> agent_definitions.AgentDefinition:
+    """Create an Nmap agent definition."""
+    return agent_definitions.AgentDefinition(
+        name="nmap",
+        in_selectors=[
+            "v3.asset.ip.v4",
+            "v3.asset.ip.v6",
+            "v3.asset.domain_name",
+            "v3.asset.link",
+        ],
+        out_selectors=["v3.asset.ip.v4.port.service"],
+        args=[
+            {
+                "name": "fast_mode",
+                "description": "Fast mode scans fewer ports than the default mode.",
+                "type": "boolean",
+                "value": True,
+            },
+            {
+                "name": "top_ports",
+                "type": "int",
+                "description": "Top ports to scan.",
+            },
+            {
+                "name": "timing_template",
+                "type": "string",
+                "description": "Template of timing settings (T0, T1, ... T5).",
+                "value": "T3",
+            },
+            {
+                "name": "scripts",
+                "type": "array",
+                "description": "List of scripts to run using Nmap",
+                "value": ["banner"],
+            },
+            {
+                "name": "float_arg",
+                "type": "number",
+                "description": "Float argument.",
+                "value": 3.14,
+            },
+        ],
+    )
+
+
+@pytest.fixture
+def run_scan_mock(mocker: plugin.MockerFixture) -> None:
+    """Mock functions required to run a scan."""
+    mocker.patch(
+        "ostorlab.cli.docker_requirements_checker.is_docker_installed",
+        return_value=True,
+    )
+    mocker.patch(
+        "ostorlab.cli.docker_requirements_checker.is_docker_working", return_value=True
+    )
+    mocker.patch(
+        "ostorlab.cli.docker_requirements_checker.is_swarm_initialized",
+        return_value=True,
+    )
+    mocker.patch("docker.from_env")
+
+    mocker.patch(
+        "ostorlab.runtimes.local.runtime.LocalRuntime.can_run", return_value=True
+    )
+    mocker.patch("ostorlab.runtimes.local.runtime.LocalRuntime._create_network")
+    mocker.patch("ostorlab.runtimes.local.runtime.LocalRuntime._start_services")
+    mocker.patch("ostorlab.runtimes.local.runtime.LocalRuntime._start_pre_agents")
+    mocker.patch("ostorlab.runtimes.local.runtime.LocalRuntime._start_agents")
+    mocker.patch("ostorlab.runtimes.local.runtime.LocalRuntime._start_post_agents")
+    mocker.patch("ostorlab.runtimes.local.runtime.LocalRuntime._check_services_healthy")
```

Comment:
Can we avoid mocking private methods? or is it too much ?

Reply:
I tried, there is no way to get over it !